mlflow/store/artifact/s3_artifact_repo.py

```diff
@@ -250,14 +250,17 @@ def delete_artifacts(self, artifact_path=None):
         if artifact_path:
             dest_path = posixpath.join(dest_path, artifact_path)
 
+        prefix = dest_path + "/" if dest_path else ""
         s3_client = self._get_s3_client()
-        list_objects = s3_client.list_objects(Bucket=bucket, Prefix=dest_path).get("Contents", [])
-        for to_delete_obj in list_objects:
-            file_path = to_delete_obj.get("Key")
-            self._verify_listed_object_contains_artifact_path_prefix(
-                listed_object_path=file_path, artifact_path=dest_path
-            )
-            s3_client.delete_object(Bucket=bucket, Key=file_path)
+        paginator = s3_client.get_paginator("list_objects_v2")
+        results = paginator.paginate(Bucket=bucket, Prefix=prefix)
+        for result in results:
+            for to_delete_obj in result.get("Contents", []):
+                file_path = to_delete_obj.get("Key")
+                self._verify_listed_object_contains_artifact_path_prefix(
+                    listed_object_path=file_path, artifact_path=dest_path
+                )
+                s3_client.delete_object(Bucket=bucket, Key=file_path)
```

Comment:
can we use [`delete_objects`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/delete_objects.html)?

Reply:
Good call, delete_objects should be able to reduce per-request overhead. Updated!